Sources/JWTKit/X5C/X5CVerifier.swift

```diff
@@ -121,13 +121,26 @@ public struct X5CVerifier: Sendable {
             return data
         }
 
-        // Setup an untrusted chain using all the certificates in the x5c.
+        // Setup an untrusted chain using all the certificates in the x5c
         let untrustedChain = try CertificateStore(certificateData.map {
             try Certificate(derEncoded: [UInt8]($0))
         })
 
+        let payload = try parser.payload(as: Payload.self, jsonDecoder: jsonDecoder)
+
+        let date: Date
+        // Some JWT implementations have the sign date in the payload.
+        // If it's such a payload, we'll use that date for validation
+        if let validationTimePayload = payload as? ValidationTimePayload {
```

Comment:
I'd still like to see if we can compare against `Payload.Type` here

Reply:
@0xTim I mean we can but to access the property afterwards we still need to cast it

Reply:
Ahh of course